R/evaluate-clusters.R

```diff
@@ -307,3 +307,94 @@ calculate_stability <- function(
 
   return(all_ari_df)
 }
+
+#' Evaluate set of clusters
+#'
+#' This wrapper function can be used to evaluate clusters calculated using `sweep_clusters()` function.
+#' Input should be be a list of data frames with the resulting clusters from all parameter combinations provided to
+#' the `sweep_clusters()` function. Output
+#'
+#' @param x An object containing PCs that clusters were calculated from. This can be
+#'   either a SingleCellExperiment object, a Seurat object, or a matrix where columns
+#'   are PCs and rows are cells. If a matrix is provided, it must have row names of cell
+#'   ids (e.g., barcodes).
+#' @param sweep_list A list of data frames obtained from `sweep_clusters()`. each data frame
+#'   in the list that contains at least two columns: one representing
+#'   unique cell ids, and one containing cluster assignments. By default, these columns
+#'   should be named `cell_id` and `cluster` respectively, though this can be customized.
+#'   The cell id column's values should match either the PC matrix row names, or the
+#'   SingleCellExperiment/Seurat object cell ids. Typically this data frame will be
+#'   output from the `rOpenScPCA::calculate_clusters()` function.
```

Comment:
```suggestion
#' @param sweep_list A list of data frames obtained from `rOpenScPCA::sweep_clusters()`. Each data frame
#'   in the list should contains at least two columns: one representing
#'   unique cell ids, and one containing cluster assignments. By default, these columns
#'   should be named `cell_id` and `cluster` respectively, though this can be customized.
#'   The cell id column's values should match either the PC matrix row names, or the
#'   SingleCellExperiment/Seurat object cell ids. 
```

Reply:
I think we want explicit arguments for these three here as well, rather than sending them in through `...`:

https://github.com/AlexsLemonade/rOpenScPCA/blob/b4aeb0c59c9305def39073bf8f44af0950aecb0c/R/evaluate-clusters.R#L40-L42